pkg/build/generator/generator.go

```diff
@@ -864,6 +864,9 @@ func generateBuildFromBuild(build *buildv1.Build, buildConfig *buildv1.BuildConf
 		},
 	}
 	// TODO remove/update this when we support cloning binary builds
+	// we need to explicitly set type to empty string so that this does not get
+	// defaulted to non-empty binary build
+	newBuild.Spec.Source.Type = ""
```

Comment:
this field is a string, how is it not always defaulted to empty string?

Reply:
We copy entire `.spec` from the previous build a couple lines above (see https://github.com/openshift/origin/blob/ebc5e89575ca09ccb48d3dfc08c67eb95ef795be/pkg/build/generator/generator.go#L854). Similarly for generating from BC we copy entire `.spec.source` (see https://github.com/openshift/origin/blob/ebc5e89575ca09ccb48d3dfc08c67eb95ef795be/pkg/build/generator/generator.go#L503).